src/uu/pidwait/src/pidwait.rs

```diff
@@ -0,0 +1,246 @@
+                } else {
+                    name.into()
+                };
+
+                let want = if settings.exact {
+                    &name
+                } else if settings.full {
+                    &it.cmdline
+                } else {
+                    &it.proc_stat()[..15]
+                };
+
+                REGEX.get().unwrap().is_match(want)
+            };
+            if matched {
+                temp.push(it)
+            }
+        }
+        temp
+    };
+
+    // Process `-O`
+    let mut proc_infos = {
+        let mut temp: Vec<ProcessInformation> = Vec::new();
+        let older = settings.older.unwrap_or_default();
+        for mut proc_info in proc_infos {
+            if proc_info.start_time().unwrap() >= older {
+                temp.push(proc_info)
+            }
+        }
+        temp
+    };
+
+    if proc_infos.is_empty() {
+        return proc_infos;
+    }
+
+    // Sorting oldest and newest
+    let proc_infos = if settings.oldest || settings.newest {
+        proc_infos.sort_by(|a, b| {
+            b.clone()
+                .start_time()
+                .unwrap()
+                .cmp(&a.clone().start_time().unwrap())
+        });
+
+        let start_time = if settings.newest {
+            proc_infos.first().cloned().unwrap().start_time().unwrap()
+        } else {
+            proc_infos.last().cloned().unwrap().start_time().unwrap()
+        };
+
+        // There might be some process start at same time, so need to be filtered.
+        let mut filtered = proc_infos
+            .iter()
+            .filter(|it| (*it).clone().start_time().unwrap() == start_time)
+            .collect::<Vec<_>>();
+
+        if settings.newest {
+            filtered.sort_by(|a, b| b.pid.cmp(&a.pid))
+        } else {
+            filtered.sort_by(|a, b| a.pid.cmp(&b.pid))
+        }
+
+        vec![filtered.first().cloned().unwrap().clone()]
+    } else {
+        proc_infos
+    };
+
+    proc_infos
+}
+
+#[allow(clippy::cognitive_complexity)]
+pub fn uu_app() -> Command {
+    Command::new(env!("CARGO_PKG_NAME"))
+        .version(crate_version!())
+        .about(ABOUT)
+        .override_usage(format_usage(USAGE))
+        .infer_long_args(true)
+        .args([
+            arg!(-e --echo                      "display PIDs before waiting"),
+            arg!(-c --count                     "count of matching processes"),
+            arg!(-f --full                      "use full process name to match"),
+            // arg!(-g --pgroup        <PGID>      "match listed process group IDs"),
+            // arg!(-G --group         <GID>       "match real group IDs"),
+            arg!(-i --"ignore-case"             "match case insensitively"),
+            arg!(-n --newest                    "select most recently started"),
+            arg!(-o --oldest                    "select least recently started"),
+            arg!(-O --older         <seconds>   "select where older than seconds"),
```

Comment:
I think you need a value parser for `--older` because currently there is an error:
```
$ cargo run -q pidwait --echo -O 1000
thread 'main' panicked at /home/dho/projects/procps/src/uu/pidwait/src/pidwait.rs:48:24:
Mismatch between definition and access of `older`. Could not downcast to u64, need to downcast to alloc::string::String
```